### PR TITLE
Jmstall/fix1511

### PR DESCRIPTION
**Pull request by MikeStall**
Fix for race condition in #1511

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/MarshalTests.cs [hunk 1]
```diff
@@ -927,5 +927,36 @@ public void DynamicMarshaller()
...
+            var engine = new Engine();
+
+            for (int i = 0; i < 100; i++)
+            {
+                var cache = new TypeMarshallerCache();
+                RecordType recordType = (RecordType)cache.GetMarshaller(typeof(MyType2)).Type;
+
+                Parallel.For(
+                    0,
+                    1,
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/MarshalTests.cs:949]: If you want to stress test it, shouldn't we have more parallel iterations?

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/MarshalTests.cs [hunk 2]
```diff
@@ -927,5 +927,36 @@ public void DynamicMarshaller()
...
+                    (j) =>
+                    {
+                        var i2 = i;
```
Comment (nguhoa) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/MarshalTests.cs:952]: this variable is never used. should you remove?

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/MarshalTests.cs (hunk 1 above)
Comment (MikeStall) [reply src/tests/Microsoft.PowerFx.Interpreter.Tests/MarshalTests.cs]: whoops, yes.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by nguhoa — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**